nautobot/extras/signals.py

```diff
@@ -286,6 +287,66 @@ def dynamic_group_membership_created(sender, instance, raw=False, **kwargs):
 pre_save.connect(dynamic_group_membership_created, sender=DynamicGroupMembership)
 
 
+def dynamic_group_eligible_groups_changed(sender, instance, **kwargs):
+    """
+    When a DynamicGroup is created or deleted, refresh the cache of eligible groups for the associated ContentType.
+
+    Can't change content_type_id on an existing instance, so no need to check for that.
+    """
+
+    if get_settings_or_config("DYNAMIC_GROUPS_MEMBER_CACHE_TIMEOUT") == 0:
+        # Caching is disabled, so there's nothing to do
+        return
+
+    if kwargs.get("created", None) is False:
+        # We do not care about updates
+        # "created" is not a kwarg for post_delete signals, so is unset or None
+        # "created" is a kwarg for post_save signals, but you cannot change content types of existing groups
+        #   therefor we can ignore cache updates on DynamicGroups updates
+        return
+
+    content_type = instance.content_type
```

Comment:
can/should we use cached content type here?

Reply:
🐛

Reply:
Reverted. This is the FK on DynamicGroup to the applicable content type (`DynamicGroup.content_type`)